vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/java/com/vaadin/flow/component/spreadsheet/Spreadsheet.java

```diff
@@ -3797,11 +3798,14 @@ protected void reloadActiveSheetData() {
         clearSheetOverlays();
         topLeftCellCommentsLoaded = false;
 
-        UI.getCurrent().beforeClientResponse(this, e -> {
-            if (reload) {
-                this.updateReloadState();
-            }
+        Optional.ofNullable(UI.getCurrent()).ifPresent(ui -> {
```

Comment:
Maybe we can change this to use `getUI().ifPresent(action);` instead.

Reply:
Done

Reply:
Needed to revert this cleanup because if broke some tests

Reply:
That's odd. Both ways should be similar. Tried running the `UndoRedoIT` locally, and it worked with the previous change, though. Weird.

Reply:
For me it started failing locally also 🤷‍♂️